app/scripts/modules/ecs/src/serverGroup/configure/serverGroupConfiguration.service.ts

```diff
@@ -280,6 +288,49 @@ export class EcsServerGroupConfigurationService {
       });
   }
 
+  public setCapacityProviderDetails(command: IEcsServerGroupCommand): void {
+    this.$q.all({
+      capacityProviderDetails: this.ecsClusterReader.listDescribeClusters(command.credentials, command.region)
+    }).then((result: Partial<IEcsServerGroupCommandBackingData>) => {
+      command.backingData.capacityProviderDetails = chain(result.capacityProviderDetails)
+        .map((cluster) => this.mapCapacityProviderDetails(cluster))
+        .value();
+    });
+
+    if(command.ecsClusterName == null || command.ecsClusterName.length == 0){
```

Comment:
nit: for ease of readability, it's usually easier to put the explicityly valid condition, as it's possible in the future to have more or conditions in which it could be false so you would have a bunch of || statements vs just having the condition that is the true case and anything else falls under else

```
if (command.ecsClusterName != null && command.ecsClusterName.length > 0) {
    this.configureAvailableCapacityProviders(command)
} else {
    command.backingData.filtered.availableCapacityProviders = [];
    command.backingData.filtered.defaultCapacityProviderStrategy = [];
}
```

Reply:
Resolved.